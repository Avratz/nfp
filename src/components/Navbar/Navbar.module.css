.nav {
  display: none;
}

@media (min-width: 1200px) {
  .nav {
    background-color: transparent;
    color: var(--neutral-4);
    width: 100%;
    height: 50px;
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 2rem;
    padding-left: 2rem;
  }

  .button {
    margin-left: auto;
    color: var(--neutral-8);
  }
  .heading,
  .lineSeparation {
    display: none;
  }

  .link {
    font-size: var(--text-base);
    line-height: var(--line-base);
  }

  .link,
  .link:visited {
    color: var(--neutral-4);
  }

  .link:hover {
    color: var(--neutral-5);
  }
}
