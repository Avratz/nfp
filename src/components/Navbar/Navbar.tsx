import { MotionButton } from '@/ui/Button/Button'
import { MotionHeading } from '@/ui/Heading/Heading'
import { MotionLink } from '@/ui/Link/Link'
import { Variants, motion } from 'framer-motion'

import styles from './Navbar.module.css'

const itemVariants: Variants = {
  open: {
    opacity: 1,
    y: 0,
    transition: { type: 'spring', stiffness: 300, damping: 24 },
  },
  closed: { opacity: 0, y: 20, transition: { duration: 0.2 } },
}
const sidebar: Variants = {
  open: {
    clipPath: 'inset(0% 0% 0% 0%)',
    transition: {
      type: 'spring',
      bounce: 0,
      duration: 0.3,
      delayChildren: 0.3,
      staggerChildren: 0.05,
    },
  },
  closed: {
    clipPath: 'inset(0% 0% 100% 0%)',
    transition: {
      type: 'spring',
      bounce: 0,
      duration: 0.3,
    },
  },
}

export function Navbar({ isOpen }: { isOpen: boolean }) {
  return (
    <motion.nav
      className={styles.nav}
      initial={false}
      animate={isOpen ? 'open' : 'closed'}
      variants={sidebar}
    >
      <motion.div className={styles.navInner}>
        <MotionHeading
          as="h3"
          size="2xl"
          layout
          variants={itemVariants}
          className={styles.heading}
        >
          Menu
        </MotionHeading>
        <MotionLink
          href="/"
          layout
          variants={itemVariants}
          className={styles.link}
        >
          Discover
        </MotionLink>
        <MotionLink
          href="/"
          layout
          variants={itemVariants}
          className={styles.link}
        >
          What we do
        </MotionLink>
        <motion.div
          className={styles.lineSeparation}
          variants={itemVariants}
        ></motion.div>
        <MotionHeading
          as="h3"
          size="2xl"
          layout
          variants={itemVariants}
          className={styles.heading}
        >
          Connect
        </MotionHeading>
        <MotionButton
          href="/"
          layout
          variants={itemVariants}
          intent="primaryDark"
        >
          Connect Wallet
        </MotionButton>
      </motion.div>
    </motion.nav>
  )
}
