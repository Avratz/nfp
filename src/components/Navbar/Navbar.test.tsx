import { render } from '@testing-library/react'
import { Navbar } from './Navbar'

describe('Navbar', () => {
  it('renders Navbar', () => {
    let { container } = render(<Navbar />)
    expect(container).toMatchSnapshot()
  })
})
