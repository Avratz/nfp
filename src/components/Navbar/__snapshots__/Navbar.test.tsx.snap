// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar 1`] = `
<div>
  <nav
    class="nav"
  >
    <hr
      class="divider vertical"
    />
    <a
      class="link link"
      href="/"
    >
      Discover
    </a>
    <a
      class="link link"
      href="/"
    >
      What we do
    </a>
    <a
      class="link button primary base button"
      href="/"
    >
      Connect Wallet
    </a>
  </nav>
</div>
`;
